### ISSUE TITLE
No compiled output files written by gulp-typescript

### ISSUE THREAD

**Issue by christiandreher** [ivogabe/gulp-typescript#348]
I am facing an issue with 'gulp-typescript' where the compilation does not produce any error, but doesn't emit any output files either. This seems to be related to #305.

I have created a repository with a MWE for another issue I was facing (but it seems to not be related). For the issue I am describing here, I have created a hotfix branch in the MWE. Here is the complete repo:
https://github.com/christiandreher/ang-exc-mwe/tree/hotfix
I will reference the relevant parts in the following.

> Is this related to gulp-typescript, and not to TypeScript?

Yes, it is definetly gulp-typescript, because...

> Is the behavior of gulp-typescript different than the behavior of tsc?

... tsc works without any issue (and without any warnings or errors either).

> There was no similar issue already opened? (Please don't bump old closed issues)

I was told to open a new issue for this. Anyway, like already mentioned, #305 is pretty similar as described by @mko.

**Expected behavior:**
`gulp compile` should compile my files and emit them to the destination path (`public` in my case. The output folder was committed, too, compiled by `tsc`).

**Actual behavior:**
Nothing happens. No output, no error, nothing

**Versions**
`npm ls --depth 0` gives the following output:

```
chris@lamda-bia:~/workspace/ang-exc-mwe$ npm ls --depth 0
myproject@0.0.0 /home/chris/workspace/ang-exc-mwe
├── @angular/common@2.0.0-rc.1
├── @angular/compiler@2.0.0-rc.1
├── @angular/core@2.0.0-rc.1
├── @angular/http@2.0.0-rc.1
├── @angular/platform-browser@2.0.0-rc.1
├── @angular/platform-browser-dynamic@2.0.0-rc.1
├── @angular/router@2.0.0-rc.1
├── @angular/router-deprecated@2.0.0-rc.1
├── @angular/upgrade@2.0.0-rc.1
├── angular2-in-memory-web-api@0.0.7
├── concurrently@2.1.0
├── del@2.2.0
├── es6-shim@0.35.1
├── gulp@3.9.1
├── gulp-sourcemaps@1.6.0
├── gulp-typescript@2.13.4
├── lite-server@2.2.0
├── reflect-metadata@0.1.3
├── rxjs@5.0.0-beta.6
├── systemjs@0.19.27
├── typescript@1.9.0-dev.20160517-1.0
├── typings@0.8.1
└── zone.js@0.6.12
```

**My gulpfile:**
https://github.com/christiandreher/ang-exc-mwe/blob/hotfix/gulpfile.ts#L21

```javascript
gulp.task('compile', () => {
    let tsResult = gulp.src('src/**/*.ts')
        .pipe(sourcemaps.init())
        .pipe(tsc(tsProject));
    return tsResult.js
        .pipe(sourcemaps.write('.'))
        .pipe(gulp.dest('public'));
});
```

**tsconfig.json**
https://github.com/christiandreher/ang-exc-mwe/blob/hotfix/tsconfig.json

```json
{
	"compilerOptions": {
		"target": "es5",
		"module": "system",
		"moduleResolution": "node",
		"sourceMap": true,
		"emitDecoratorMetadata": true,
		"experimentalDecorators": true,
		"removeComments": false,
		"noImplicitAny": false,
		"rootDir": "./src",
		"outDir": "./public",
		"baseUrl": ".",
		"paths": {
			"myproject/*": ["./src/myproject/*"]
		}
	},
	"exclude": [
		"gulpfile.ts",
		"node_modules",
		"public",
		"typings/main",
		"typings/main.d.ts"
	]
}
```

**Comment by adidahiya** [comment]
I have a similar issue:

- I'm pointing gulp-typescript to tsconfig.json and using `tsProject.src()`
- using both `rootDir` and `outDir` options
- `tsc -p` compiles fine
- gulp-typescript outputs nothing in the pipe once I invoke the compiler

**Comment by sonofmaw** [comment]
Looks like #299 has regressed due to a fix for #247 

Using convertCompilerOptionsFromJson overrides gulp-typescript's own conversion of the tsconfig options, including resolving baseUrl to an absolute path, which is a requirement for the paths configuration to work.

**Comment by nemenos** [comment]
I had to downgrade to 2.12 for this issue.
Problem is present regardless of typescript@next. I'm using stable release, and using outdir makes gulp-typescript fail silently. tsc works.

**Comment by ZoolWay** [comment]
I got a similar issue. With `baseUrl` in `tsconfig.json` I don't geht any output, neither `js` stream nor `dts` steam. Without the option it works but obviously I need the option for relative module resolution. Can I somehow work around it?

**Comment by ZoolWay** [comment]
I would like to share my current solution to the relative path in `baseUrl` with you until this is fixed:

```
gulp.task('build', function () {
    if (!typescriptCompiler) {
        var tsconfigJson = JSON.parse(fs.readFileSync('tsconfig.json'));
        typescriptCompiler = typescript.createProject('tsconfig.json', {
            "typescript": require('typescript'),
            "baseUrl": path.resolve('.', tsconfigJson.compilerOptions.baseUrl) // we must override with absolute path to have gulp-typescript working correctly
        });
        console.log("tsconfig.json 'baseUrl' value '%s' was resolved to '%s'", tsconfigJson.compilerOptions.baseUrl, typescriptCompiler.input.options.baseUrl);
    }
    var tsCompilerResult = 
        gulp.src([source])
            .pipe(plumber())
            .pipe(count('## source files collected'))
            .pipe(sourcemaps.init({ loadMaps: true }))
            .pipe(typescript(typescriptCompiler));

    return merge([
        tsCompilerResult.dts
            .pipe(count('## declarations created'))
            .pipe(gulp.dest(paths.dtsOutput)),
        tsCompilerResult.js
            .pipe(sourcemaps.write('.', { includeContent: false, sourceRoot: '/src' }))
            .pipe(count('## javascript files created', {logFiles: false}))
            .pipe(gulp.dest(paths.output))
    ]);
});
```

**Comment by christiandreher** [comment]
Great @ZoolWay ! You are my hero, this works perfect :))

**Comment by ivogabe** [comment]
Is this issue still present in version 3.0?

**Comment by christiandreher** [comment]
I can't confirm this, since I moved to an other toolchain, I'm sorry

**Comment by nemenos** [comment]
seems to be working properly on 3.0.2

**Comment by jkrot** [comment]
I am having this same or similar problem in     "gulp-typescript": "^3.1.6",
I had to manually declare the root url paths.

**Comment by BMalaichik** [comment]
@jkrot can you please share your solution?

**Comment by jkrot** [comment]
gulp file 
```
var ts = require("gulp-typescript");
var tsProject  = ts.createProject('tsconfig.json');
```

tsconfig.js
```
{
    "compilerOptions": {
        "allowJs": false,
        "target": "es5",
        "removeComments": true,
        "isolatedModules": false,
        "rootDir": "./app",
        "outDir": "./app/js/build",
        "baseUrl": ".",
        "diagnostics": true
    },
    "files": [
        "./node_modules/@types/angular/index.d.ts",
        "./node_modules/@types/angular-resource/index.d.ts",
        "./node_modules/@types/angular-route/index.d.ts",
        "./node_modules/@types/angular-sanitize/index.d.ts",
        "./app/main.module.ts"
    ],
    "include": [
        "./app/**/*.ts"
    ],
    "exclude": [
        "node_modules",
        "./app/**/*.spec.js",
        "./app/**/*.spec.ts"
    ]

}
```

package.json

```
  },
  "devDependencies": {
    "angular-mocks": "1.6.4",
    "del": "^2.2.2",
    "express": "^4.15.2",
    "gulp": "^3.9.1",
    "gulp-angular-templatecache": "^2.0.0",
    "gulp-autoprefixer": "^3.1.1",
    "gulp-clean-css": "^3.0.4",
    "gulp-concat": "^2.6.1",
    "gulp-conventional-changelog": "^1.1.3",
    "gulp-css-base64": "^1.3.4",
    "gulp-csslint": "^1.0.0",
    "gulp-cssmin": "^0.1.7",
    "gulp-debug": "^3.1.0",
    "gulp-file-include": "^1.0.0",
    "gulp-flatten": "^0.3.1",
    "gulp-footer": "^1.0.5",
    "gulp-header": "^1.8.8",
    "gulp-livereload": "^3.8.1",
    "gulp-markdown": "^1.2.0",
    "gulp-plumber": "^1.1.0",
    "gulp-rename": "^1.2.2",
    "gulp-sourcemaps": "^2.5.0",
    "gulp-tap": "^0.1.3",
    "gulp-typescript": "^3.1.6",
    "gulp-uglify": "^2.1.2",
    "gulp-util": "^3.0.8",
    "gulp-watch": "^4.3.11",
    "jasmine": "^2.5.3",
    "karma": "^1.5.0",
    "karma-chrome-launcher": "^2.0.0",
    "karma-cli": "^1.0.1",
    "karma-coverage": "^1.1.1",
    "karma-htmlfile-reporter": "^0.3.5",
    "karma-jasmine": "^1.1.0",
    "karma-spec-reporter": "^0.0.26",
    "lazypipe": "^1.0.1",
    "merge2": "^1.0.3",
    "node-fs": "^0.1.7",
    "protractor": "5.1.1",
    "st": "^1.2.0",
    "typescript": "^2.2.2",
    "typings": "^2.1.1"  },
  "dependencies": {
    "angular": "^1.6.4",
    "@types/angular": "^1.6.14",
    "angular-resource": "^1.6.4",
    "@types/angular-resource": "^1.5.8",
    "angular-route": "^1.6.4",
    "@types/angular-route": "^1.3.3",
    "angular-sanitize": "^1.6.4",
    "@types/angular-sanitize": "^1.3.4",
    "angular-translate": "^2.15.1"
  }
```


I couldn't get any of the other things to work with types so I brought in all the types via npm and manually build them.  This is the only way I could get all of this to work.
